Tests/JExtractSwiftTests/JNI/JNIClosureTests.swift

```diff
@@ -0,0 +1,129 @@
+      detectChunkByInitialLines: 1,
+      expectedChunks: [
+        """
+        @_cdecl("Java_com_example_swift_SwiftModule__00024emptyClosure__Lcom_example_swift_SwiftModule_00024emptyClosure_00024closure_2")
+        func Java_com_example_swift_SwiftModule__00024emptyClosure__Lcom_example_swift_SwiftModule_00024emptyClosure_00024closure_2(environment: UnsafeMutablePointer<JNIEnv?>!, thisClass: jclass, closure: jobject) {
+          SwiftModule.emptyClosure(closure: { 
+            let class$ = environment!.interface.GetObjectClass(environment, closure)
+            let methodID$ = environment!.interface.GetMethodID(environment, class$, "apply", "()V")!
+            let arguments$: [jvalue] = []
+            environment!.interface.CallVoidMethodA(environment, closure, methodID$, arguments$)
+          }
+          )
+        }
+        """
+      ]
+    )
+  }
+
+  @Test
+  func closureWithArgumentsAndReturn_javaBindings() throws {
+    try assertOutput(input: source, .jni, .java, expectedChunks: [
+      """
+      public static class closureWithArgumentsAndReturn {
+        @FunctionalInterface
+        public interface closure {
+          long apply(long _0, boolean _1);
+        }
+      }
+      """,
+      """
+      /**
+        * Downcall to Swift:
+        * {@snippet lang=swift :
+        * public func closureWithArgumentsAndReturn(closure: (Int64, Bool) -> Int64)
+        * }
+        */
+      public static void closureWithArgumentsAndReturn(com.example.swift.SwiftModule.closureWithArgumentsAndReturn.closure closure) {
+        SwiftModule.$closureWithArgumentsAndReturn(closure);
+      }
+      """,
+      """
+      private static native void $closureWithArgumentsAndReturn(com.example.swift.SwiftModule.closureWithArgumentsAndReturn.closure closure);
+      """
+    ])
+  }
+
+  @Test
+  func closureWithArgumentsAndReturn_swiftThunks() throws {
+    try assertOutput(
+      input: source,
+      .jni,
+      .swift,
+      detectChunkByInitialLines: 1,
+      expectedChunks: [
+        """
+        @_cdecl("Java_com_example_swift_SwiftModule__00024closureWithArgumentsAndReturn__Lcom_example_swift_SwiftModule_00024closureWithArgumentsAndReturn_00024closure_2")
+        func Java_com_example_swift_SwiftModule__00024closureWithArgumentsAndReturn__Lcom_example_swift_SwiftModule_00024closureWithArgumentsAndReturn_00024closure_2(environment: UnsafeMutablePointer<JNIEnv?>!, thisClass: jclass, closure: jobject) {
+          SwiftModule.closureWithArgumentsAndReturn(closure: { _0, _1 in
+            let class$ = environment!.interface.GetObjectClass(environment, closure)
+            let methodID$ = environment!.interface.GetMethodID(environment, class$, "apply", "(JZ)J")!
+            let arguments$: [jvalue] = [_0.getJValue(in: environment!), _1.getJValue(in: environment!)]
+            return Int64(fromJNI: environment!.interface.CallLongMethodA(environment, closure, methodID$, arguments$), in: environment!)
+          }
```

Comment:
I'd kinda want to do one `guard` for the environment like we did elsewhere but this is minor, let's revisit holistically someday 👍